kpi/serializers/v2/deployment.py

```diff
@@ -53,21 +53,10 @@ def update(self, instance, validated_data):
                 {'backend': 'This field cannot be modified after the initial '
                             'deployment.'})
 
-        if ('version_id' in validated_data and
-                validated_data['version_id'] != deployment.version_id):
-            # Request specified a `version_id` that differs from the current
-            # deployment's; redeploy
-            self._raise_unless_current_version(asset, validated_data)
-            asset.deploy(
-                backend=deployment.backend,
-                active=validated_data.get('active', deployment.active)
-            )
-        elif 'active' in validated_data:
-            active = validated_data['active']
-            # Set the `active` flag without touching the rest of the deployment
-            deployment.set_active(active)
-            # If we (re)activate the asset, let's synchronize its media files
-            if active:
-                asset.sync_media_files_async(always=False)
+        self._raise_unless_current_version(asset, validated_data)
+        asset.deploy(
+            backend=deployment.backend,
+            active=validated_data.get('active', deployment.active)
+        )
```

Comment:
Changing this behaviour updates `date_deployed` even when we archive a project. Is it really what we want? 
IMO, `date_deployed` should not change on archiving?

Reply:
Good point, although I think changing `date_deployed` when (un)archiving is a lesser evil compared to *not* changing `date_deployed` when someone adds or removes a form media file and hits "REDEPLOY", which is the behavior without this PR.

I'll see about updating `date_deployed` only when the form version changes or when there are form media files to synchronize.

Reply:
Simpler approach: when the PATCH request contains only `active` (and optionally `backend`), do not update `date_deployed` :ok_hand: